components/suggestion/suggestion_box.jsx

```diff
@@ -252,19 +252,27 @@ export default class SuggestionBox extends React.Component {
         const pretext = textbox.value.substring(0, textbox.selectionEnd);
 
         let prefix;
+        let keepPretext = false;
         if (pretext.endsWith(matchedPretext)) {
             prefix = pretext.substring(0, pretext.length - matchedPretext.length);
         } else {
             // the pretext has changed since we got a term to complete so see if the term still fits the pretext
             const termWithoutMatched = term.substring(matchedPretext.length);
             const overlap = SuggestionBox.findOverlap(pretext, termWithoutMatched);
```

Comment:
This works fine until you have delays with the server.

Certain situations can occur where the suggestion term is invalid which would cause weird behaviors like;
`@user`, when typed would output `@u@u@user`

Therefore, only apply text substitution when there's overlap.  If not `keepPretext` is set to `true`